### PR TITLE
[ROR 1.26.0][RORDEV-269] response_fields rule

### PR DESCRIPTION
**Pull request by pielas**
I see that atom removed unnecessary spaces from this document. Let me know if that's ok

### CODE REVIEW

File: elasticsearch.md [hunk 1]
```diff
@@ -920,6 +920,46 @@ After adding the `filter` rule \(using the block duplication strategy\).
       kibana_index: ".kibana_@{user}"
 ```
 
+#### `response_fields`
+
+This rule allows filtering Elasticsearch responses using defined list of fields. It works in very similar way to fields rule. However it **doesn't make use of Field Level Security \(FLS\)** and could be applied to every response returned by Elasticsearch.
```
Comment (coutoPL) [inline elasticsearch.md:925]: to the `fields` rule

File: elasticsearch.md [hunk 2]
```diff
@@ -920,6 +920,46 @@ After adding the `filter` rule \(using the block duplication strategy\).
...
+
+It could be configured in two modes: *whitelist* passing only defined fields or *blacklist* filtering out defined fields.
```
Comment (coutoPL) [inline elasticsearch.md:927]: maybe bullet list for the modes?

File: elasticsearch.md (hunk 1 above)
Comment (coutoPL) [inline elasticsearch.md:925]: maybe we should mention that in contrast to `fields` rule which filters out document fields, this rule filters our response fields?

File: elasticsearch.md [hunk 3]
```diff
@@ -920,6 +920,46 @@ After adding the `filter` rule \(using the block duplication strategy\).
...
+
+**Blacklist mode**
+
+Specifies which fields should be filtered out with `~` (other fields from mapping become allowed implicitly). Example:
+
+`fields: ["~excluded_fields_prefix_*", "~excluded_field", "~another_excluded_field.nested_field"]`
+
+Return documents but deprived of the fields that:
```
Comment (coutoPL) [inline elasticsearch.md:935]: let's don't talk about document in this rule. This rule filters response fields, not document fields (like `fields` rule)

File: elasticsearch.md [hunk 4]
```diff
@@ -920,6 +920,46 @@ After adding the `filter` rule \(using the block duplication strategy\).
...
+  * are equal to `excluded_field`
+  * are equal to `another_excluded_field.nested_field`
+
+**Whitelist mode**
+
+In this mode rule is configured to filter out each field that isn't defined in rule.
+
+`response_fields: ["allowed_fields_prefix_*", "_*", "allowed_field.nested_field.text"]`
+
+Return documents deprived of all the fields, except the ones that:
```
Comment (coutoPL) [inline elasticsearch.md:946]: same here

File: elasticsearch.md [hunk 5]
```diff
@@ -920,6 +920,46 @@ After adding the `filter` rule \(using the block duplication strategy\).
...
+
+*Example*: allow only `cluster_name` and `status` field in cluster health response:
+
+```
+- name: "Filter cluster health response"
+  uri_re: "^/_cluster/health"
+  response_fields: ["cluster_name", "status"]
+```
+
+Any field could be filtered using this rule.
```
Comment (coutoPL) [inline elasticsearch.md:961]: maybe we should show an example of full `/_cluster/health`response body and the filtered one?

**Review by coutoPL — CHANGES_REQUESTED** [review]

**Comment by coutoPL** [comment]
@pielas please set `develop` as a target branch
@sscarduzio please also review the PR and change default branch to develop

**Review by coutoPL — CHANGES_REQUESTED** [review]
@pielas what has happened? I see whole file changed? There is really hard to figure out what has changed

**Comment by pielas** [comment]
> @pielas what has happened? I see whole file changed? There is really hard to figure out what has changed

Atom somehow changed new line characters, this is why there was change in whole file. I fixed it and squashed my commits with review fixes

**Review by coutoPL — APPROVED** [review]
LGTM. @sscarduzio do you want to take a look?

File: elasticsearch.md [hunk 6]
```diff
@@ -920,6 +920,76 @@ After adding the `filter` rule \(using the block duplication strategy\).
       kibana_index: ".kibana_@{user}"
 ```
 
+#### `response_fields`
+
+This rule allows filtering Elasticsearch responses using defined list of fields. It works in very similar way to `fields` rule. However in contrast to `fields` rule, which filters out document fields, this rule filters out response fields. It **doesn't make use of Field Level Security \(FLS\)** and could be applied to every response returned by Elasticsearch.
```
Comment (sscarduzio) [inline elasticsearch.md:925]: "using a list of fields".

Comment (sscarduzio) [inline elasticsearch.md:925]: Remove that "However".

Comment (sscarduzio) [inline elasticsearch.md:925]: ~could~  -> can

 (apply also below.)

File: elasticsearch.md [hunk 7]
```diff
@@ -920,6 +920,76 @@ After adding the `filter` rule \(using the block duplication strategy\).
...
+
+It could be configured in two modes:
+  * *whitelist* passing only defined fields
+  * *blacklist* filtering out defined fields.
```
Comment (sscarduzio) [inline elasticsearch.md:929]: filtering out (removing) only the defined fields from the response JSON object

File: elasticsearch.md [hunk 8]
```diff
@@ -920,6 +920,76 @@ After adding the `filter` rule \(using the block duplication strategy\).
...
+
+**Blacklist mode**
+
+Specifies which fields should be filtered out with `~` (other fields from mapping become allowed implicitly). Example:
```
Comment (sscarduzio) [inline elasticsearch.md:933]: filtered out by adding the `~` prefix to the field name. Other fields in the response will be implicitly allowed. For example:

File: elasticsearch.md [hunk 9]
```diff
@@ -920,6 +920,76 @@ After adding the `filter` rule \(using the block duplication strategy\).
...
+
+`fields: ["~excluded_fields_prefix_*", "~excluded_field", "~another_excluded_field.nested_field"]`
+
+Return response but deprived of the fields that:
```
Comment (sscarduzio) [inline elasticsearch.md:937]: The above will return the usual response object, but deprived (if found) of the fields that:

File: elasticsearch.md [hunk 10]
```diff
@@ -920,6 +920,76 @@ After adding the `filter` rule \(using the block duplication strategy\).
...
+
+It could be configured in two modes:
+  * *whitelist* passing only defined fields
+  * *blacklist* filtering out defined fields.
+
+**Blacklist mode**
+
+Specifies which fields should be filtered out with `~` (other fields from mapping become allowed implicitly). Example:
+
+`fields: ["~excluded_fields_prefix_*", "~excluded_field", "~another_excluded_field.nested_field"]`
```
Comment (sscarduzio) [inline elasticsearch.md:935]: "fields" -> "respone_fields"

File: elasticsearch.md [hunk 11]
```diff
@@ -920,6 +920,76 @@ After adding the `filter` rule \(using the block duplication strategy\).
...
+
+Return response but deprived of the fields that:
+  * start with `excluded_fields_prefix_`
+  * are equal to `excluded_field`
+  * are equal to `another_excluded_field.nested_field`
+
+**Whitelist mode**
+
+In this mode rule is configured to filter out each field that isn't defined in rule.
```
Comment (sscarduzio) [inline elasticsearch.md:944]: in THE rule

File: elasticsearch.md [hunk 12]
```diff
@@ -920,6 +920,76 @@ After adding the `filter` rule \(using the block duplication strategy\).
...
+```
+response from above will look like:
+```
+{
+	"cluster_name": "ROR_SINGLE",
+	"status": "yellow"
+}
+```
+
+**NB:** Any field could be filtered using this rule.
```
Comment (sscarduzio) [inline elasticsearch.md:991]: Any response field can be filtered using this rule.

**Review by sscarduzio — CHANGES_REQUESTED** [review]
Some changes for clarity + one typo

File: elasticsearch.md [hunk 13]
```diff
@@ -920,6 +920,76 @@ After adding the `filter` rule \(using the block duplication strategy\).
       kibana_index: ".kibana_@{user}"
 ```
 
+#### `response_fields`
+
+This rule allows filtering Elasticsearch responses using defined list of fields. It works in very similar way to `fields` rule. However in contrast to `fields` rule, which filters out document fields, this rule filters out response fields. It **doesn't make use of Field Level Security \(FLS\)** and could be applied to every response returned by Elasticsearch.
+
+It could be configured in two modes:
+  * *whitelist* passing only defined fields
```
[the remark this answers is not in the document]
Comment (pielas) [reply elasticsearch.md:928]: I removed "JSON" because this rule can also filter yaml or binary cbor responses

File: elasticsearch.md (hunk 7 above)
Comment (pielas) [reply elasticsearch.md:929]: I removed "JSON" because this rule can also filter yaml or binary cbor responses

**Pull request merged**